tests/stack/test_api.py

```diff
@@ -529,3 +578,69 @@ def test_download_resource(self, stack_api):
             api.RESOURCE_PROPERTIES: {'test': 'test'}
         }
         stack_api._download_resource(resource_badservice)
+
+    def test_deploy_valid_stack_status(self, stack_api, tmpdir):
+        """
+            The stack status generated from a correctly set up stack passed through deployment
```

Comment:
The test name "test_deploy_valid_stack_status" is not clear, how about just "test_deploy_config", because this is the main method tested here.
Also could you improve the docstring? It seems that "deploy_config method should generate correct status" is suffice.